realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,741 @@
+    @Test
+    @RunTestInLooperThread
+    public void notification_notSentAfterUnregisterListenerModelObject() {
+        final Realm looperThreadRealm = looperThread.realm;
+
+        looperThreadRealm.beginTransaction();
+        AllJavaTypes child = looperThreadRealm.createObject(AllJavaTypes.class, 10);
+        looperThreadRealm.commitTransaction();
+
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                fail("Not expecting notification after unregister");
+            }
+        };
+        child.addChangeListener(listener);
+        child.removeChangeListener(listener);
+
+        looperThreadRealm.beginTransaction();
+        AllJavaTypes parent = looperThreadRealm.createObject(AllJavaTypes.class, 1);
+        parent.setFieldObject(child);
+        looperThreadRealm.commitTransaction();
+
+        verifyPostConditions(
+            looperThreadRealm,
+            new PostConditions() {
+                public void run(Realm realm) {
+                    assertEquals(2, looperThreadRealm.where(AllJavaTypes.class).findAll().size());
+                }
+            },
+            child, parent);
+    }
+
+    // A listener registered on the backlinked object should be called when a backlinked object is deleted
+    @Test
+    @RunTestInLooperThread
+    public void notification_onDeleteModelObject() {
+        final Realm looperThreadRealm = looperThread.realm;
+
+        looperThreadRealm.beginTransaction();
+        AllJavaTypes child = looperThreadRealm.createObject(AllJavaTypes.class, 10);
+        AllJavaTypes parent = looperThreadRealm.createObject(AllJavaTypes.class, 1);
+        parent.setFieldObject(child);
+        looperThreadRealm.commitTransaction();
+
+        final AtomicInteger counter = new AtomicInteger(0);
+        RealmChangeListener<AllJavaTypes> listener = new RealmChangeListener<AllJavaTypes>() {
+            @Override
+            public void onChange(AllJavaTypes object) {
+                counter.incrementAndGet();
+            }
+        };
+        child.addChangeListener(listener);
+
+        looperThreadRealm.beginTransaction();
+        looperThreadRealm.where(AllJavaTypes.class).equalTo("fieldId", 1).findAll().deleteAllFromRealm();
+        looperThreadRealm.commitTransaction();
+
+        verifyPostConditions(
+            looperThreadRealm,
+            new PostConditions() {
+                public void run(Realm realm) {
+                    assertEquals(1, looperThreadRealm.where(AllJavaTypes.class).findAll().size());
+                    assertEquals(1, counter.get());
+                }
+            },
+            child, parent);
+    }
+
+    // A listener registered on the backlinked object is called for an unrelated change on the an object of the same type!!
+    // This test exists only to document existing (but odd) behavior.
```

Comment:
Better to  remove this test in #4101, documenting odd behaviour with a unit test is nice I think, since we then detect if it changes, like the "notification-for-deleted-objects" we suddenly got for free from the OS.